cognite/powerops/resync/v2/shop_to_assets.py

```diff
@@ -0,0 +1,370 @@
+        plant_time_series_mappings: Optional[list] = None,
+        watercourses: Optional[list] = None,
+    ):
+        self.shop_model_by_directory = shop_model_by_directory
+        self.times_series_by_generator_name = {
+            entry["generator_name"]: {k: str(v) for k, v in entry.items()}
+            for entry in generator_times_series_mappings or []
+        }
+        self.times_series_by_plant_name = {
+            entry["plant_name"]: {k: str(v) for k, v in entry.items()} for entry in plant_time_series_mappings or []
+        }
+        self.watercourse_by_directory = {entry["directory"]: entry for entry in watercourses or []}
+
+    @classmethod
+    def from_directory(cls, directory: Path, file_name: str = "model_raw") -> PowerAssetImporter:
+        shop_model_files = list(directory.glob(f"**/{file_name}.yaml"))
+        shop_model_by_watercourse = {
+            file.parent.name: load_yaml(file, expected_return_type="dict") for file in shop_model_files
+        }
+
+        generator_mapping_file = directory / "generator_time_series_mappings.yaml"
+        generator_mappings = (
+            load_yaml(generator_mapping_file, expected_return_type="list") if generator_mapping_file.exists() else None
+        )
+
+        plant_mapping_file = directory / "plant_time_series_mappings.yaml"
+        plant_mappings = (
+            load_yaml(plant_mapping_file, expected_return_type="list") if plant_mapping_file.exists() else None
+        )
+
+        watercourses_file = directory / "watercourses.yaml"
+        watercourses = load_yaml(watercourses_file, expected_return_type="list") if watercourses_file.exists() else None
+
+        return cls(shop_model_by_watercourse, generator_mappings, plant_mappings, watercourses)
+
+    def to_power_assets(self) -> list[GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite]:
+        assets_by_xid: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite] = {}
+        for watercourse_dir, shop_model in self.shop_model_by_directory.items():
+            watercourse_assets = self._model_to_assets(shop_model, watercourse_dir, assets_by_xid)
+            assets_by_xid.update(watercourse_assets)
+
+        return list(assets_by_xid.values())
+
+    def _model_to_assets(
+        self,
+        shop_model: dict[str, Any],
+        watercourse_dir: str,
+        existing: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite],
+    ) -> dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite]:
+        assets_by_xid: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite] = {}
+        try:
+            watercourse_config = self.watercourse_by_directory[watercourse_dir]
+        except KeyError as e:
+            raise ValueError(f"Watercourse directory {watercourse_dir} not watercourses.yaml file") from e
```

Comment:
**polish:** don't think this error message is very clear as to what the issue is?
```suggestion
            raise ValueError(f"Watercourse directory {watercourse_dir} does not exist in the watercourses.yaml configuration") from e
```